### PR TITLE
add course-service

### PR DESCRIPTION
**Pull request by yyc1217**
累死我惹...

### CODE REVIEW

File: course-service/course/course.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+                </tr>
+                <tr>
+                    <td>isPreSelect</td>
+                    <td>Boolean</td>
+                    <td>will be selected for student before first selection.</td>
+                </tr>
+                <tr>
+                    <td>teachers</td>
+                    <td>String</td>
+                    <td>seperated by ","</td>
```
Comment (jason1122g) [inline course-service/course/course.md]: separated

File: course-service/course/limit.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	<td><i>no</i></td>
+	<td>year + semester
+		<table>
+			<tr>
+				<td>year</td>
+				<td>3 digits of MinGuo year</td>
+			</tr>
+			<tr>
+				<td>semester</td>
+				<td>1 digit of semester, 1 for top and 2 for bottom.</td>
```
Comment (jason1122g) [inline course-service/course/limit.md]: first semester
second semester

File: course-service/course/limit.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/course/limit.md]: Field

File: course-service/search/department.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/search/department.md]: Field

File: course-service/search/search.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
+# Search Course Information
+
+```
+GET /search
+```
+
+## Description
+> Search course information at current semseter.
```
Comment (jason1122g) [inline course-service/search/search.md]: semester

File: course-service/search/search.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/search/search.md]: Field

File: course-service/search/summer.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/search/summer.md]: Field

File: course-service/search/target.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/search/target.md]: Field

File: course-service/status.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/status.md]: Field

File: course-service/student/selected.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/student/selected.md]: Field

File: course-service/student/tracking.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/student/tracking.md]: Field

File: course-service/unit/college.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/unit/college.md]: Field

File: course-service/unit/college_department.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/unit/college_department.md]: Field

File: course-service/unit/department_target.md [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+
+# Response
+
+## Formats
+- json
+
+## Structure
+<table>
+    <tr>
+		<td><b>Filed Name</b></td>
```
Comment (jason1122g) [inline course-service/unit/department_target.md]: Field

**Comment by yyc1217** [comment]
@jason1122g done~

**Pull request merged**